src/evo/specialtxman.cpp

```diff
@@ -53,18 +53,11 @@ static bool CheckSpecialTxInner(CDeterministicMNManager& dmnman, const Chainstat
             }
             return CheckMNHFTx(chainman, qman, tx, pindexPrev, state);
         case TRANSACTION_ASSET_LOCK:
-            if (!DeploymentActiveAfter(pindexPrev, consensusParams, Consensus::DEPLOYMENT_V20)) {
```

Comment:
why keeping similar checks for "unlock" and "mnhf" if we drop this one?

Reply:
@UdjinM6 don't we normally keep checks like this? for stuff like new devnets, or in case someone is sending us a bad fork I guess?

Reply:
what will happen if someone will send us a bad fork for devnet? If fork will be accepted - that's fine, whatever it has asset-unlock/mnehf/etc transaction or not. we have chainlocks also.

That's important to keep conditions like that if they change behavior incompatible way. For example DIP0001 has not only introduced a bigger blocks size, but also introduced a limit for size of transaction and some old blocks do not satisfy that requirement. So, for that type of consensus condition we should keep it, otherwise mainnet/testnet can't re-sync properly for old blocks. But there's completely fine. 

Can you show in our code base any condition like that which we keep intentionally over-versions?

Reply:
also, we have all soft-forks on all devnets activated from blocks 300 (soon from block 2), so, it should not be a problem also
https://github.com/dashpay/dash/pull/6187/files#diff-ff53e63501a5e89fd650b378c9708274df8ad5d38fcffa6c64be417c4d438b6dL568-L573

Reply:
with headers-first sync and assumevalid in place these checks do not matter that much anymore and we can be drop them safely imo